src/conn/conn_api.c

```diff
@@ -2505,6 +2505,16 @@ __conn_version_verify(WT_SESSION_IMPL *session)
 
     conn = S2C(session);
 
+    /*
+     * Initialize the recovery version variables. Later these will be updated with proper WiredTiger
+     * version if the database is being opened for recovery from the WiredTiger.turtle file.
+     * WiredTiger recovery occurs not only when the database is restarted after a clean or unclean
+     * shutdown, but also when the backup is restored.
+     */
+    conn->recovery_major = 0;
+    conn->recovery_minor = 0;
+    conn->recovery_patch = 0;
```

Comment:
This change is unnecessary. The connection structure is guaranteed to be zeroed on allocation. There is a lot of code that depends on this. Having this here is confusing because it causes me to wonder what they were set to that this is needed/here. I suggest removing this.

Reply:
I aksed for something like this. I had to trace through the code to get certainty that it'll work. I'm OK to have a different solution, but the implicit relationship where not retrieving a version when the turtle file doesn't exist implies a 0 release, and we rely on that for future decision making feels fragile.

Reply:
It is only looked at in recovery code so we know it is during wiredtiger_open where we've allocated a connection. Maybe a comment in the recovery code saying "this was set up when we read the turtle file in <function>." and then in that function add a comment that it is set here or will be zero if there is no turtle file from a backup (or whatever the details of when we do/do not are).

Reply:
As an alternative, we can initialize those variables in all the places where we don't call `__wt_turtle_validate_version`.

Reply:
`__conn_version_verify` is the first function that checks for the turtle file, I feel initializing it here may be enough.